common/persistence/sql/queue_v2.go

```diff
@@ -43,7 +43,7 @@ var (
 // NewQueueV2 returns an implementation of [persistence.QueueV2] which always returns an error because it is not
 // implemented yet.
 func NewQueueV2() persistence.QueueV2 {
-	return &queueV2{}
+	return queueV2{}
```

Comment:
the `queueV2` struct will contain other fields later after it's implemented I think? Then I think still make sense to return a pointer and methods need to be declared with pointer receivers? We can change later but I am afraid that might be missed. Or you can add some comments here.